qiime2_TagSequencingReport.Rmd

```diff
@@ -0,0 +1,383 @@
+### ================
+###   R packages
+### ================
+library(plyr)
+library(dplyr)
+library(qiimer)
+library(pander)
+library(ape)
+library(vegan)
+library(ggplot2)
+library(tidyr)
+library(broom)
+library(reshape2)
+library(kylemisc)
+library(gtools) 
+```
+
+<!-- resources -->
+```{r resources}
+### ================
+###   R resources
+### ================
+source("R_functions.R")
+```
+
+<!-- user defined functions -->
+```{r user defined functions}
+### ================
+###   User defined functions, if any
+### ================
+```
+
+<!-- ================================================================================================ -->
+<!--   End of Preamble                                                                                -->
+<!-- ================================================================================================ -->
+
+<!-- ================================================================================================ -->
+<!--   Beginning of Project Specific Items                                                            -->
+<!-- ================================================================================================ -->
+
+```{r}
+### ===========================
+###   define constants
+### ===========================
+
+### run date
+run_date <- "October 3rd, 2017"
+
+### investigator
+investigator <- "Mark Brown"
+
+### data files directory
+data_files_dir <- "data_files"
+
+### porject directory
+proj_dir <- "16s"
+
+### mapping file
+mapping_file_fp <- file.path(data_files_dir, "20171003_brown_mapping_file.txt")
+
+### per-sample read count 
+read_count_fp <- file.path(proj_dir, "table", "sample-frequency-detail.csv")
+
+### otu table
+feature_table_fp <- file.path(proj_dir, "table", "feature-table.tsv")
+
+### taxonomic assignment 
+taxo_assignment_fp <- file.path(proj_dir, "taxonomy", "taxonomy.tsv")
+
+### unweighted UniFrac
+uu_fp <- file.path(proj_dir, "core-metrics-results", "uu", "distance-matrix.tsv")
+
+### weighted UniFrac
+wu_fp <- file.path(proj_dir, "core-metrics-results", "wu", "distance-matrix.tsv")
+
+### num of observed otu
+notu_fp <- file.path(proj_dir, "core-metrics-results", "observed_otu", "alpha-diversity.tsv")
+
+### shannon
+shannon_fp <- file.path(proj_dir, "core-metrics-results", "shannon", "alpha-diversity.tsv")
+
+### minimum reads threshold
+min_reads <- 1000
+```
+
+
+```{r}
+### ===========================
+###   read in data
+### ===========================
+
+### read mapping file
+s <- read_qiime_mapping_file(mapping_file_fp) 
+
+### Check if the sample names start with a number. If they do, add an "X" to the front
+toFix = grepl("^[0-9]", s$SampleID)
+s$SampleIDorig = s$SampleID
+s$SampleID[toFix] = paste0('X', s$SampleID[toFix])
+
+### read otu table
+o <- read.table(feature_table_fp, skip=1, header=T, as.is=T, sep="\t", comment.char="", fill=T)
+rownames(o) <- o$X.OTU.ID
+counts <- o %>%
+  select(-X.OTU.ID, -taxonomy)
```

Comment:
We are storing the OTU counts here as a `data.frame`, rather than a matrix.  Not sure if this messes up anything down the line, but something to watch out for.  I am more familiar with the matrix format, and I believe `vegan` likes to see a matrix.

Generally, we are at a crossroads with the count information.  On one hand, we can decide to use the Erik Clarke method and use one giant data frame with one row per OTU and sample.  Alternately, we can stick with the matrix format and operate like we have.  For now, probably best to stay the course and stick with the matrix.

Reply:
counts is now stored as a matrix and noticed no problems.